warehouse/packaging/__init__.py

```diff
@@ -49,11 +60,24 @@ def includeme(config):
     config.register_origin_cache_keys(
         User,
         cache_keys=["user/{obj.username}"],
+    )
+    config.register_origin_cache_keys(
+        User.name,
```

Comment:
@dstufft Adding these lines are removing the purge keys from `User` and adding the same ones for `User.name`. Sorry I didn't split this into two commits, that may have been more clear.